### DISCUSSION TITLE
Foundry Fundamentals lesson 14: Issue: Error Message "No such file or directory (os error 2)" when deploying smart contract locally using Anvil

### DISCUSSION THREAD

**Discussion by mehan05** [Cyfrin/foundry-full-course-cu#1516] [Q&A] [answered]
When attempting to deploy a smart contract locally using Anvil, the following command was used:

`forge script script/<filename.s.sol>:<contractName> --rpc-url http:127.0.0.1:8545 --broadcast  --private-key <private_key>
`

**But I got this error:**

```
Error: 
No such file or directory (os error 2)
```
What does this error mean? 
Help me to find out this error,  And Thanks in Advance

**Comment by EngrPips** [comment, marked as answer]
Hello @MEHAN-M-2005 , the command didn't work because you didn't put the required value in the placeholders that are in the command . so let me  explain what i mean below 

`forge script script/<here you would put the name of the script you want to run>:<here you would put the actual contract you want to run in that script> --rpc-url http:127.0.0.1:8545 --broadcast --private-key <here you would put your private key>`

<img width="1440" alt="Screenshot 2024-03-23 at 1 23 35 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/ab772483-3097-43c6-aeeb-94e48017c341">

The above is an image of my vscode showing a project repo, now let's say i want to run the `DeployRaffle.s.sol` script and inside that script i want to run the `DeployRaffle` contract. assuming my private key is `0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80` . So the command i would run is below ;

`forge script script/DeployRaffle.s.sol:DeployRaffle --rpc-url http:127.0.0.1:8545 --broadcast --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80` 

Note: make sure to have started a local anvil chain in terminal byt running `anvil` .

This should work.

**Comment by mehan05** [comment]
Hi [EngrPips](https://github.com/EngrPips), Thanks for helping me I did what you said, And the error is gone, Thank you😊😊
![Screenshot 2024-03-23 145555](https://github.com/Cyfrin/foundry-full-course-f23/assets/115608700/c6c69e19-be64-4869-99c4-b12b6f3fe0fd)

**Reply by EngrPips** [reply]
Hello @MEHAN-M-2005 , i am glad i could help and maybe you could please mark this discussion as answered.
